src/complex/Utilities/SIMPLConversion.hpp

```diff
@@ -0,0 +1,2466 @@
+      row[0] = ParameterType::k_CrystalStructures[iter[k_CrystalStructureKey].get<int32>()];
+      row[1] = ParameterType::k_PhaseTypes[iter[k_PhaseTypeKey].get<int32>()];
+      row[2] = iter[k_PhaseNameKey].get<std::string>();
+
+      value.push_back(std::move(row));
+    }
+
+    return {std::move(value)};
+  }
+};
+
+struct GenerateColorTableFilterParameterConverter
+{
+  using ParameterType = GenerateColorTableParameter;
+  using ValueType = ParameterType::ValueType;
+
+  static constexpr StringLiteral k_SelectedPresetKey = "SelectedPresetName";
+  static constexpr StringLiteral k_PresetNameKey = "Name";
+  static constexpr StringLiteral k_RGBPointsKey = "RGBPoints";
+
+  static Result<ValueType> convert(const nlohmann::json& json1, const nlohmann::json& json2)
+  {
+    if(!json1.is_string())
+    {
+      return MakeErrorResult<ValueType>(-2, fmt::format("GenerateColorTableFilterParameterConverter json1 '{}' is not a string", json2.dump()));
+    }
+    if(!json2.is_array())
+    {
+      return MakeErrorResult<ValueType>(-1, fmt::format("GenerateColorTableFilterParameterConverter json2 '{}' is not an array", json1.dump()));
+    }
+
+    auto presetName = json1.get<std::string>();
+    auto rgbPoints = json2.get<std::vector<float64>>();
+
+    nlohmann::json value = nlohmann::json::object();
+    value[k_PresetNameKey] = presetName;
+    value[k_RGBPointsKey] = rgbPoints;
+
+    return {std::move(value)};
+  }
+};
+
+struct ReadASCIIWizardDataFilterParameterConverter
+{
+  using ParameterType = ReadCSVFileParameter;
+  using ValueType = ParameterType::ValueType;
+
+  static constexpr StringLiteral k_InputFilePathKey = "Wizard_InputFilePath";
+  static constexpr StringLiteral k_DataHeadersKey = "Wizard_DataHeaders";
+  static constexpr StringLiteral k_BeginIndexKey = "Wizard_BeginIndex";
+  static constexpr StringLiteral k_NumberOfLinesKey = "Wizard_NumberOfLines";
+  static constexpr StringLiteral k_DataTypesKey = "Wizard_DataTypes";
+  static constexpr StringLiteral k_DelimitersKey = "Wizard_Delimiters";
+  static constexpr StringLiteral k_HeaderLineKey = "Wizard_HeaderLine";
+  static constexpr StringLiteral k_HeaderIsCustomKey = "Wizard_HeaderIsCustom";
+  static constexpr StringLiteral k_HeaderUsesDefaultsKey = "Wizard_HeaderUseDefaults";
+  static constexpr StringLiteral k_ConsecutiveDelimitersKey = "Wizard_ConsecutiveDelimiters";
+
+  static std::vector<DataType> ConvertDataTypeStrings(const std::vector<std::string>& dataTypes)
+  {
+    std::vector<DataType> output;
+
+    for(usize i = 0; i < dataTypes.size(); i++)
+    {
+      try
+      {
+        output.push_back(complex::StringToDataType(dataTypes[i]));
+      } catch(const std::exception& e)
+      {
+      }
+    }
+
+    return output;
+  }
+
+  static std::vector<char> ConvertToChars(const std::string& string)
+  {
+    std::vector<char> chars(string.length());
+    for(usize i = 0; i < string.size(); i++)
+    {
+      chars[i] = string[i];
+    }
+    return chars;
```

Comment:
```suggestion
    return std::vector<char>(string.begin(), string.end());
```
Just use the iterator constructor for std::vector